### PR TITLE
MTV-6640 | fix(openstack): assign stable disk index by source volume position

### PR DESCRIPTION
**Pull request by krcmarik**
Cold migrations of multi-disk OpenStack VMs consistently failed on 5.0 with `persistentvolumeclaims "<vm>-disk-0" already exists` during the CreateDataVolumes phase.

PopulatorVolumes derived each disk's PVC index from a running counter that was only advanced for images already in Active status, and getImagesFromVolumes drops volumes whose Glance image/properties are not ready yet. A disk's index therefore depended on which other disks happened to be ready at reconcile time, so a slow-to-snapshot disk (e.g. a large boot disk) caused another disk to take its index and then collide once it finally became Active. This was harmless in 2.12 (random-suffix PVC names) but fatal under deterministic naming (pvcNameTemplateUseGenerateName: false).

Derive the disk index from the source volume's ordinal position in workload.Volumes (a stable, complete, ordered slice) so it stays deterministic across reconciles regardless of image readiness. The VM-snapshot image is placed after all volume disks at len(workload.Volumes) so it never collides with a volume disk index.

Add regression tests covering out-of-order/lagging volume readiness and the VM-snapshot image path.

Resolves: MTV-6640

### CODE REVIEW

File: pkg/controller/plan/adapter/openstack/builder_populator_test.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+			Properties: map[string]interface{}{
+				forkliftPropertyOriginalImageID: workload.ImageID,
+			},
+		}
+
+		images := map[string]*model.Image{
+			getImageFromVolumeName(builder.Context, workload.ID, "vol-a"): imageForVolume("vol-a"),
+			getImageFromVolumeName(builder.Context, workload.ID, "vol-b"): imageForVolume("vol-b"),
+			getVmSnapshotName(builder.Context, workload.ID):               snapshotImage,
+		}
```
[the remark this answers is not in the document]
Comment (krcmarik) [reply pkg/controller/plan/adapter/openstack/builder_populator_test.go:209]: Skipping: additional edge-case test coverage is out of scope for this fix; keeping the change minimal per the fix's intent. The existing snapshot-path test (ImageID set + 2 volumes, asserting the snapshot lands at index 2 = len(workload.Volumes)) already covers the deterministic indexing behavior.
